adapters/repos/db/aggregator/numerical.go

```diff
@@ -108,7 +108,8 @@ func (a *numericalAggregator) buildPairsFromCounts() {
 	a.pairs = append(a.pairs, make([]floatCountPair, 0, len(a.valueCounter))...)
 
 	for value, count := range a.valueCounter {
-		if count > a.maxCount {
+		// get one with higher count or lower value if counts are equal
+		if count > a.maxCount || (count == a.maxCount && value < a.mode) {
```

Comment:
Would you add a small unittest in adapters/repos/db/aggregator/numerical_test.go?

Reply:
added